libcontainer/specconv/spec_linux.go

```diff
@@ -29,26 +31,149 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-var namespaceMapping = map[specs.LinuxNamespaceType]configs.NamespaceType{
-	specs.PIDNamespace:     configs.NEWPID,
-	specs.NetworkNamespace: configs.NEWNET,
-	specs.MountNamespace:   configs.NEWNS,
-	specs.UserNamespace:    configs.NEWUSER,
-	specs.IPCNamespace:     configs.NEWIPC,
-	specs.UTSNamespace:     configs.NEWUTS,
-	specs.CgroupNamespace:  configs.NEWCGROUP,
+var (
+	initMapsOnce            sync.Once
+	namespaceMapping        map[specs.LinuxNamespaceType]configs.NamespaceType
+	mountPropagationMapping map[string]int
+	recAttrFlags            map[string]struct {
+		clear bool
+		flag  uint64
+	}
+	mountFlags, extensionFlags map[string]struct {
+		clear bool
+		flag  int
+	}
+)
+
+func initMaps() {
+	initMapsOnce.Do(func() {
+		namespaceMapping = map[specs.LinuxNamespaceType]configs.NamespaceType{
+			specs.PIDNamespace:     configs.NEWPID,
+			specs.NetworkNamespace: configs.NEWNET,
+			specs.MountNamespace:   configs.NEWNS,
+			specs.UserNamespace:    configs.NEWUSER,
+			specs.IPCNamespace:     configs.NEWIPC,
+			specs.UTSNamespace:     configs.NEWUTS,
+			specs.CgroupNamespace:  configs.NEWCGROUP,
+			specs.TimeNamespace:    configs.NEWTIME,
+		}
+
+		mountPropagationMapping = map[string]int{
+			"rprivate":    unix.MS_PRIVATE | unix.MS_REC,
+			"private":     unix.MS_PRIVATE,
+			"rslave":      unix.MS_SLAVE | unix.MS_REC,
+			"slave":       unix.MS_SLAVE,
+			"rshared":     unix.MS_SHARED | unix.MS_REC,
+			"shared":      unix.MS_SHARED,
+			"runbindable": unix.MS_UNBINDABLE | unix.MS_REC,
+			"unbindable":  unix.MS_UNBINDABLE,
+			"":            0,
```

Comment:
Why is this line (L70) needed?

Reply:
Ah got it; let's leave it there then. I asked because I don't see it in `runc`, will remove it from sysbox-runc in a follow up PR.